lib/haskell/natural4/src/LS/XPile/MathLang/GenericMathLang/GenericMathLangAST.hs

```diff
@@ -108,18 +118,25 @@ data ExplnAnnot = MkExplnAnnot
   } deriving stock (Eq, Ord, Show)
 makeFieldLabelsNoPrefix ''ExplnAnnot
 
+-- | Source position.
+--
+-- (Row and column, but not filename.)
+--
 data SrcPositn = MkPositn
-  { row :: Int
-  , col :: Int
+  { row :: !Int
```

Comment:
It's not clear to me if there's a simple rule of thumb we can follow for when to make data strict -- might be worth mentioning to the group at some point, or even just leaving a quick comment in Slack, if there's some such simple rule of thumb?

Reply:
It's tricky.

But as an extremely extremely general rule of thumb go with:

- flat, unstructured, (constant-sized) data should by default be strict;
- structured, tree-shaped data should by default be lazy.

So this means that record fields of type `Int`, `Double`, `Bool` are good candidates for adding `!`s.